pandas/tests/copy_view/test_replace.py

```diff
@@ -122,17 +123,22 @@ def test_replace_inplace(using_copy_on_write):
         assert df._mgr._has_no_reference(0)
 
 
-@pytest.mark.parametrize("to_replace", [1.5, [1.5]])
+@pytest.mark.parametrize("to_replace", [1.5, [1.5], []])
 def test_replace_inplace_reference(using_copy_on_write, to_replace):
     df = DataFrame({"a": [1.5, 2, 3]})
     arr_a = get_array(df, "a")
     view = df[:]
     df.replace(to_replace=to_replace, value=15.5, inplace=True)
 
     if using_copy_on_write:
-        assert not np.shares_memory(get_array(df, "a"), arr_a)
-        assert df._mgr._has_no_reference(0)
-        assert view._mgr._has_no_reference(0)
+        if isinstance(to_replace, list) and len(to_replace) == 0:
```

Comment:
Might be better to parametrize ``test_replace_emptylist`` for ``inplace=True/False``

Reply:
added it to another test where it added less complexity